### PR TITLE
Add master operations to mesos input plugin

### PR DESCRIPTION
**Pull request by philipnrmn**
A recent Mesos release shipped new metrics for operations on masters:
https://reviews.apache.org/r/70116

This commit adds all those metrics to the mesos input plugin.

Screenshot from 2019-06-24 16-29-19

### CODE REVIEW

**Comment by philipnrmn** [comment]
Test failure on this PR is a flake in dcos_containers.

**Comment by philipnrmn** [comment]
DC/OS PR: https://github.com/dcos/dcos/pull/5761

File: plugins/inputs/mesos/README.md [hunk 1]
```diff
@@ -142,6 +143,44 @@ Mesos master metric groups
     - master/tasks_staging
     - master/tasks_starting
 
+- operations
+    - master/operations/total
+		- master/operations/pending
```
Comment (gkleiman) [inline plugins/inputs/mesos/README.md]: I think that the suffixes here (`total`, `pending`, `recoverable`, `unreachable`) don't match the ones that Mesos exposes.

[Here](https://github.com/apache/mesos/blob/e5f48794f0354f8179f4302b308fb89a9f7caf7c/src/master/metrics.cpp#L606-L614) is the code that generates the metrics, and [here's](https://mesos.apache.org/documentation/latest/monitoring/#operations) the documentation describing which metrics are counters and which are gauges.

The gist is that Mesos uses gauges for the non-terminal states (`pending`, `recovering`, or `unreachable`) and counters for the terminal states (`finished`, `error`, `dropped`, or `gone_by_operator`).

There are also three more metrics exposed by the master: `master/valid_operation_status_updates`, `master/invalid_operation_status_updates`, and `master/messages_operation_status_update`.

Comment (philipnrmn) [reply plugins/inputs/mesos/README.md]: Thanks! I addressed your comments.

File: plugins/inputs/mesos/mesos.go [hunk 2]
```diff
@@ -134,6 +134,7 @@ var sampleConfig = `
     "frameworks",
     "framework_offers",
     "tasks",
+		"operations",
```
Comment (gracedo) [inline plugins/inputs/mesos/mesos.go]: whitespace looks off here

**Review by gracedo — COMMENTED** [review]
Correct me if I'm wrong, but I think for variables in the metric name, we try to extrapolate those out into tags (I recall us doing this for the master framework offers and allocator metrics when we added them in).

Following the same pattern as the framework offers and allocator metrics, it would look something like:
- `master/operations/total` -> `master/operations_total`
- `master/operations/<OPERATION_STATE>` -> `master/operations` with additional `operation_state: <OPERATION_STATE>` tag
- `master/operations/<OPERATION_TYPE>/total` -> `master/operations/total` with additional `operation_type: <OPERATION_TYPE>` tag
- `master/operations/<OPERATION_TYPE>/<OPERATION_STATE>` -> `master/operations` with additional `operation_type: <OPERATION_TYPE>` and `operation_state: <OPERATION_STATE>` tags

It doesn't resolve quite as cleanly as the other framework offers and allocator metrics however. Would it make sense here to do the same?

**Comment by philipnrmn** [comment]
@gracedo I did originally implement this as tags, exactly as you suggested. After some thought, I backed it out and implemented as it is now, because the set of possible metrics is known and relatively small. 

Elsewhere, we only use tags where we the set of possible metrics is very large and we cannot match all possible metrics (eg framework name).

File: plugins/inputs/mesos/mesos.go (hunk 2 above)
Comment (philipnrmn) [reply plugins/inputs/mesos/mesos.go]: Thanks! upfixed :)

**Review by gracedo — APPROVED** [review]

**Review by branden — APPROVED** [review]

**Pull request merged**